### PR TITLE
enable unittests

### PR DESCRIPTION
**Pull request by nimrof**
Unittests work now, but the tests fail.
So need to fix the tests before merge

### CODE REVIEW

**Comment by nimrof** [comment]
@trojanobelix @trojanobelix  Open for review, just do not merge this one until after #82 is merged and all checks are hopefully green :)

**Review by trojanobelix — APPROVED** [review]

**Pull request merged**
